akka-docs/src/main/paradox/stream/operators/Source/maybe.md

```diff
@@ -14,9 +14,35 @@ Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if comp
 
 ## Description
 
-Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if completed with a @scala[`Some[T]`] @java[`Optional`] 
-will emit that *T* and then complete the stream, or if completed with @scala[`None`] @java[`empty Optional`] complete the stream right away.
+Materialize a @scala[`Promise[Option[T]]`] @java[`CompletableFuture`] which controls what element will be emitted by the
+Source.
 
+* If the materialized promise is completed with a @scala[`Some`]@java[non-empty `Optional`],
+  that value will be produced downstream, followed by completion.
+* If the materialized promise is completed with a @scala[`None`]@java[empty `Optional`],
+  no value will be produced downstream and completion will be signalled immediately.
+* If the materialized promise is completed with a failure, then the returned source will terminate with that error.
```

Comment:
Do we use "terminate" elsewhere? To me the word sounds like it covers both successful and failure. Should it be "source will fail" instead?